plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/glue/IcebergGlueCatalogModule.java

```diff
@@ -53,7 +53,7 @@ protected void setup(Binder binder)
         // Required to inject HiveMetastoreFactory for migrate procedure
         binder.bind(Key.get(boolean.class, HideDeltaLakeTables.class)).toInstance(false);
         newOptionalBinder(binder, Key.get(new TypeLiteral<Predicate<Table>>() {}, ForGlueHiveMetastore.class))
-                .setBinding().toProvider(IcebergGlueMetastoreTableFilterProvider.class);
+                .setBinding().toInstance(table -> true);
```

Comment:
as a follow-up we should replace `Predicate<Table>` bidngin with a named interface.